near-sdk/tests/store_performance_tests.rs

```diff
@@ -0,0 +1,391 @@
+            .max_gas()
+            .transact()
+            .await;
+
+        let res = txn?;
+        let total_gas = res.unwrap().total_gas_burnt.as_gas();
+        perform_asserts(total_gas, col);
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn contains() -> anyhow::Result<()> {
+    // Each collection gets the same number of elements.
+    let element_number = 100;
+    let (account, contract_id) = setup().await?;
+    // Vector does not implement contains.
+    let collection_filter = |col: &Collection| !matches!(col, Collection::Vector);
+    // prepopulate
+    for col in Collection::iter().filter(collection_filter) {
+        let txn = account
+            .call(&contract_id, "insert")
+            .args_json((col, DEFAULT_INDEX_OFFSET, element_number))
+            .max_gas()
+            .transact()
+            .await;
+
+        let res = txn?;
+        let _ = res.unwrap();
+    }
+
+    // contains test, repeat here is the number of times we check all the elements in each collection.
+    // It's used to measure relative performance.
+    for (col, repeat) in Collection::iter().filter(collection_filter).map(|col| match col {
+        Collection::TreeMap => (col, 12),
+        Collection::IterableSet => (col, 11),
+        Collection::IterableMap => (col, 12),
+        Collection::UnorderedSet => (col, 11),
+        Collection::UnorderedMap => (col, 12),
+        Collection::LookupMap => (col, 16),
+        Collection::LookupSet => (col, 14),
+        _ => (col, 0),
+    }) {
+        let txn = account
+            .call(&contract_id.clone(), "contains")
+            .args_json((col, repeat, element_number))
+            .max_gas()
+            .transact()
+            .await;
+
+        let res = txn?;
+        let total_gas = res.unwrap().total_gas_burnt.as_gas();
+
+        perform_asserts(total_gas, col);
+    }
+
+    Ok(())
+}
+
+// This test demonstrates the difference in gas consumption between iterable and unordered collections,
+// when most of the elements have been deleted.
+#[tokio::test]
+async fn iterable_vs_unordered() -> anyhow::Result<()> {
+    let element_number = 300;
+    let deleted_element_number = 299;
+    let (account, contract_id) = setup().await?;
+
+    // We only care about Unordered* and Iterable* collections.
+    let collection_filter = |col: &Collection| {
+        matches!(
+            col,
+            Collection::UnorderedSet
+                | Collection::UnorderedMap
+                | Collection::IterableMap
+                | Collection::IterableSet
+        )
+    };
+
+    // insert `element_number` elements.
+    for (col, max_iterations) in
+        Collection::iter().filter(collection_filter).map(|col| (col, element_number))
+    {
+        let txn = account
+            .call(&contract_id, "insert")
+            .args_json((col, DEFAULT_INDEX_OFFSET, max_iterations))
+            .max_gas()
+            .transact()
+            .await;
+
+        let _ = txn?.unwrap();
+    }
```

Comment:
Is there something that blocks us from significantly simplifying the code?

```suggestion
    // We only care about Unordered* and Iterable* collections.
    let collection_types = &[
        Collection::UnorderedSet,
        Collection::UnorderedMap,
        Collection::IterableMap,
        Collection::IterableSet,
    ];

    // insert `element_number` elements.
    for col in &collection_types {
        account
            .call(&contract_id, "insert")
            .args_json((col, DEFAULT_INDEX_OFFSET, element_number))
            .max_gas()
            .transact()
            .await??;
    }
```